spec/controllers/course_summaries_controller_spec.rb

```diff
@@ -46,6 +48,30 @@
         end
         expect(response.status).to eq(200)
       end
+
+      context 'tests the second csv row which contains out of values' do
+        it 'checks the out of row is numerically and sequentially correct' do
+          # this test checks that the following
+          # 1) the out_row is the correct length (right amount of assessments included)
+          # 2) the order of each assessment in the header corresponds to the right value in the out_of row
+          # 3) the values in the out_of row correspond to the CORRECT values of the respective assessments
+          assignments = create_list(:assignment_with_criteria_and_results, 3)
+          grade_forms = create_list(:grade_entry_form_with_data, 2)
+          create(:grouping_with_inviter_and_submission, assignment: assignments[0])
+          create(:grouping_with_inviter, assignment: assignments[0])
+          csv_rows = get_as(@admin, :download_csv_grades_report, format: :csv).parsed_body
+          header = csv_rows[0]
+          out_of_row = csv_rows[1]
+          expect(out_of_row.size).to eq(4 + assignments.size + grade_forms.size)
+          expect(out_of_row.size).to eq(header.size)
+          zipped_info = Assessment.all.order(:id).zip(out_of_row[4, out_of_row.size])
+          zipped_info.each do |tup|
```

Comment:
This is the right use of zip, but you can do a bit better by using tuple unpacking. You can write

```ruby
zipped_info.each do |model, out_of_element|
  ...
```

so that the variables are automatically bound to the first and second element of each tuple.